dotCMS/src/main/java/com/dotcms/rest/MapToContentletPopulator.java

```diff
@@ -332,46 +339,99 @@ public List<Category> getCategories (final Contentlet contentlet, final User use
+        List<Category> categories = new ArrayList<>();
+        final List<com.dotcms.contenttype.model.field.Field> fields = contentlet.getContentType()
+                .fields(CategoryField.class);
+        for (final com.dotcms.contenttype.model.field.Field field: fields) {
+            final Map<String, Object> map = contentlet.getMap();
+            final Object object = map.get(field.variable());
+            if(object instanceof Collection){
+                final Collection<?> list = (Collection<?>) object;
+                final String joinedCategories = list.stream().map(Object::toString).collect(
+                        Collectors.joining(","));
+                buildUpCategories(categories, joinedCategories, user, respectFrontendRoles);
+                continue;
+            }
+            if(object instanceof String){
+                buildUpCategories(categories, (String)object, user, respectFrontendRoles);
+                continue;
+            }
+            if (object == null && map.containsKey(field.variable()) ){
+                //something was set to null a cat on purpose
+                //The requirement says that null must not even be considered
+                //We should only remove a category if an empty array is passed
+                map.remove(field.variable());
+                continue;
+            }
+            if(object == null){
+                //the value wasn't set. Just ignore it.
+                continue;
             }
-        }
 
+            throw new IllegalArgumentException(String.format(
+                    "Attempt to set unrecognized object [%s] as category. This field only accepts arrays like ['cat-key','cat-var','cat-inode'] or a string like 'cat-key','cat-var','cat-inode' ",
+                    object));
+        }
         return categories;
     }
 
+    /**
+     * given a List and a string representation of a category
+     * This will try to find different criterion to get the respective category
+     * @param categories
+     * @param stringValue
+     * @param user
+     * @param respectFrontendRoles
+     */
+    private void buildUpCategories(final List<Category> categories, final String stringValue,
+            final User user, boolean respectFrontendRoles) {
+        if (UtilMethods.isSet(stringValue)) {
+            final CategoryAPI categoryAPI = APILocator.getCategoryAPI();
+            for (final String categoryValue : stringValue.split("\\s*,\\s*")) {
+
+                    // try it as catId
+                    Category category = Try.of(()->categoryAPI.find(categoryValue, user, respectFrontendRoles)).getOrNull();
+                    if (category != null && InodeUtils.isSet(category.getCategoryId())) {
+                        categories.add(category);
+                        Logger.info(MapToContentletPopulator.class, String.format(" value [%s] resolved as a valid Category-Inode.", categoryValue));
```

Comment:
should be info? isn't too noise

Reply:
honestly, I don't think there will be a ton of requests like this to generate such noise. 
In fact, the line appears just once per request. And it's kinda helpful

Reply:
I'm not sure @fabrizzio-dotCMS if it's true using a custom dataset. This log is printed for each category and internally this method is called by ContentResource and WorkflowResource

Reply:
I lowered  the log leve to debug